AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -377,7 +396,8 @@ - (void)_lazilyLoadImageIfNecessary
 - (void)asyncdisplaykit_asyncTransactionContainerStateDidChange
 {
   if (self.asyncdisplaykit_asyncTransactionContainerState == ASAsyncTransactionContainerStateNoTransactions) {
-    if (self.layer.contents != nil && [self.delegate respondsToSelector:@selector(imageNodeDidFinishDecoding:)]) {
+    ASDN::MutexLocker l(_lock);
+    if (self.layer.contents != nil && _delegateSupportsImageNodeDidFinishDecoding) {
       [self.delegate imageNodeDidFinishDecoding:self];
```

Comment:
I was thinking about that too, but to be truly thread safe, we need to, otherwise the delegate *could* change between when we get the value in the lock and when we call the delegate method. It probably would never happen, but…